pkg/kubelet/rocket/rocket.go

```diff
@@ -0,0 +1,590 @@
+	if err := r.RunPod(pod, volumeMap); err != nil {
+		return err
+	}
+	return nil
+}
+
+// KillContainer kills the container in the given pod.
+// Like RunContainerInPod, we will have to tear down the whole pod first to kill this
+// single container.
+func (r *Runtime) KillContainerInPod(container api.Container, pod *api.Pod) error {
+	if err := r.KillPod(pod); err != nil {
+		return err
+	}
+
+	// Update the pod and start it.
+	var containers []api.Container
+	for _, c := range pod.Spec.Containers {
+		if c.Name == container.Name {
+			continue
+		}
+		containers = append(containers, c)
+	}
+	pod.Spec.Containers = containers
+	// TODO(yifan): Bug here, since we cannot get the volume map, after killing the mount
+	// path will disappear. This could be fixed if we support killing single container without
+	// tearing down the whole pod.
+	return r.RunPod(pod, nil)
+}
+
+// RunCommand invokes 'rkt' and returns the result from stdout
+// in a list of strings.
+// TODO(yifan): Do not export this.
+func (r *Runtime) RunCommand(args ...string) ([]string, error) {
+	glog.V(4).Info("Run rkt command:", args)
+
+	cmd := exec.Command(rocketBinName)
+	cmd.Args = append(cmd.Args, r.config.parseGlobalOptions()...)
+	cmd.Args = append(cmd.Args, args...)
+
+	output, err := cmd.Output()
+	if err != nil {
+		return nil, err
+	}
+	return strings.Split(strings.TrimSpace(string(output)), "\n"), nil
+}
+
+type podInfo struct {
+	state       string
+	networkInfo string
+	created     time.Time
+}
+
+// getIP returns the IP of a pod by parsing the network info.
+// The network info looks like this:
+//
+// default:ip4=172.16.28.3, database:ip4=172.16.28.42
+//
+func (p *podInfo) getIP() string {
+	parts := strings.Split(p.networkInfo, ",")
+
+	for _, part := range parts {
+		if strings.HasPrefix(part, "default:") {
+			return strings.Split(part, "=")[1]
+		}
+	}
+	return ""
+}
+
+// getContainerStatus converts the rocket pod state to the api.containerStatus.
+// TODO(yifan): Get more detailed info such as Image, ImageID, etc.
+func (p *podInfo) getContainerStatus() api.ContainerStatus {
+	var status api.ContainerStatus
+	switch p.state {
+	case podStateRunning:
+		// TODO(yifan): Get StartedAt.
+		status.State = api.ContainerState{
+			Running: &api.ContainerStateRunning{
+				StartedAt: util.NewTime(p.created),
+			},
+		}
+	case podStateEmbryo, podStatePreparing, podStatePrepared:
+		status.State = api.ContainerState{Waiting: &api.ContainerStateWaiting{}}
+	case podStateExited, podStateDeleting, podStateGone:
+		status.State = api.ContainerState{
+			Termination: &api.ContainerStateTerminated{
+				StartedAt: util.NewTime(p.created),
+			},
+		}
+	default:
+		glog.Warningf("Unknown pod state: %q", p.state)
+	}
+	return status
+}
+
+func (p *podInfo) toPodStatus(pod *kubecontainer.Pod) api.PodStatus {
+	var status api.PodStatus
+	status.PodIP = p.getIP()
+	// For now just make every container's state as same as the pod.
```

Comment:
@jonboulle Not yet IMO. Since if one container dies, all remaining containers will die for now, so it's ok to set all container's state to the same. But might need to show individual's state in the future if we can update one single container.